File: Delphi/SmtpClientSSL/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, clMailMessage, clTcpClient, clTcpClientTls, clMC, clSmtp, clCertificate,
  clTcpCommandClient, DemoBaseFormUnit, ExtCtrls;

type
  TForm1 = class(TclDemoBaseForm)
    Label1: TLabel;
    edtServer: TEdit;
    Label2: TLabel;
    edtPort: TEdit;
    Label3: TLabel;
    edtUser: TEdit;
    Label4: TLabel;
    edtPassword: TEdit;
    Label5: TLabel;
    edtFrom: TEdit;
    Label6: TLabel;
    edtTo: TEdit;
    Label7: TLabel;
    edtSubject: TEdit;
    memBody: TMemo;
    cbUseTLS: TCheckBox;
    btnSend: TButton;
    clSmtp1: TclSmtp;
    clMailMessage1: TclMailMessage;
    procedure btnSendClick(Sender: TObject);
    procedure clSmtp1VerifyServer(Sender: TObject; ACertificate: TclCertificate; const AStatusText: string; AStatusCode: Integer;
      var AVerified: Boolean);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.btnSendClick(Sender: TObject);
begin
  if (clSmtp1.Active) then Exit;

  clSmtp1.Server := edtServer.Text;
  clSmtp1.Port := StrToInt(edtPort.Text);

  clSmtp1.UserName := edtUser.Text;
  clSmtp1.Password := edtPassword.Text;

  if cbUseTLS.Checked then
  begin
    clSmtp1.UseTLS := ctAutomatic;
  end else
  begin
    clSmtp1.UseTLS := ctNone;
  end;

  clSmtp1.Open();
  try
    clMailMessage1.BuildMessage(memBody.Text, '');
    clMailMessage1.From.FullAddress := edtFrom.Text;
    clMailMessage1.ToList.EmailAddresses := edtTo.Text;
    clMailMessage1.Subject := edtSubject.Text;

    clSmtp1.Send(clMailMessage1);

    ShowMessage('The message was sent successfully.');
  finally
    clSmtp1.Close();
  end;
end;

procedure TForm1.clSmtp1VerifyServer(Sender: TObject; ACertificate: TclCertificate; const AStatusText: string;
  AStatusCode: Integer; var AVerified: Boolean);
begin
  if not AVerified and (MessageDlg(AStatusText + #13#10' Do you wish to proceed ?',
    mtWarning, [mbYes, mbNo], 0) = mrYes) then
  begin
    AVerified := True;
  end;
end;

end.

File: CBuilder/SmtpClientSSL/unit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "unit1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "DemoBaseFormUnit"
#pragma link "clMailMessage"
#pragma link "clMC"
#pragma link "clSmtp"
#pragma link "clTcpClient"
#pragma link "clTcpClientTls"
#pragma link "clTcpCommandClient"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
  : TclDemoBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnSendClick(TObject *Sender)
{
  if (clSmtp1->Active) return;

  clSmtp1->Server = edtServer->Text;
  clSmtp1->Port = StrToInt(edtPort->Text);

  clSmtp1->UserName = edtUser->Text;
  clSmtp1->Password = edtPassword->Text;

  if (cbUseTLS->Checked) {
    clSmtp1->UseTLS = ctAutomatic;
  }
  else {
    clSmtp1->UseTLS = ctNone;
  }

  clSmtp1->Open();
  __try {
    clMailMessage1->BuildMessage(memBody->Text, "");
    clMailMessage1->From->FullAddress = edtFrom->Text;
    clMailMessage1->ToList->EmailAddresses = edtTo->Text;
    clMailMessage1->Subject = edtSubject->Text;

    clSmtp1->Send(clMailMessage1);

    ShowMessage("The message was sent successfully.");
  }
  __finally {
    clSmtp1->Close();
  }
}
//---------------------------------------------------------------------------

void __fastcall TForm1::clSmtp1VerifyServer(TObject *Sender,
      TclCertificate *ACertificate, const UnicodeString AStatusText,
      int AStatusCode, bool &AVerified)
{
  if (!AVerified && (MessageDlg(AStatusText + "\r\n Do you wish to proceed ?",
    mtWarning, TMsgDlgButtons() << mbYes << mbNo, 0) == mrYes)) {
    AVerified = true;
  }
}
//---------------------------------------------------------------------------

